catalogs/tests/catalogs.spec.js

```diff
@@ -0,0 +1,63 @@
+describe('Component: Catalogs', function(){
+    var scope,
+        q,
+        oc,
+        catalog;
+    beforeEach(module(function($provide){
+        $provide.value('Parameters', {search:null, page: null, pageSize: null, searchOn: null, sortBy: null, userID: null, userGroupID: null, level: null, buyerID: null})
+    }));
+    beforeEach(module('orderCloud'));
+    beforeEach(module('orderCloud.sdk'));
+    beforeEach(inject(function($rootScope, $q, OrderCloud){
+        scope = $rootScope.$new();
+        q = $q;
+        oc = OrderCloud;
+        catalog = {
+            ID: 'CatalogID',
+            Name: 'CatalogName'
+        }
+    }));
+
+    describe('State: catalogs', function(){
+        var state;
+        beforeEach(inject(function($state, OrderCloudParameters){
+            state = $state.get('catalogs', {}, {reload: true});
+            spyOn(OrderCloudParameters, 'Get').and.returnValue(null);
+            spyOn(oc.Catalogs, 'List').and.returnValue(null);
+        }));
+        it('should resolve Parameters', inject(function($injector, OrderCloudParameters){
+            $injector.invoke(state.resolve.Parameters);
+            expect(OrderCloudParameters.Get).toHaveBeenCalled();
+        }));
+        it('should resolve CatalogsList', inject(function($injector){
+            $injector.invoke(state.resolve.CatalogsList);
+            expect(oc.Catalogs.List).toHaveBeenCalled();
```

Comment:
We always want to use toHaveBeenCalledWith over, just toHaveBeenCalled. So if you can think of a way to mock some parameters and expect them, then I would do that. For example on CatalogsList you can mock Parameters. One good thing to check is that if you don't set a pageSize it should still be set to 12